core/eolearn/core/utils/testing.py

```diff
@@ -80,3 +83,15 @@ def _get_feature_shape(
 ) -> Tuple[int, ...]:
     time, height, width, depth = len(timestamps), *config.raster_shape, rng.integers(*config.depth_range)
     return (time, height, width, depth) if ftype.is_temporal() else (height, width, depth)
+
+
+def assert_feature_match(tested_feature: Any, expected_feature: Any) -> None:
+    if type(tested_feature) is not type(expected_feature):
+        assert type(tested_feature) is type(expected_feature)
+
+    if isinstance(tested_feature, np.ndarray):
+        assert_array_equal(tested_feature, expected_feature)
+    elif isinstance(tested_feature, gpd.GeoDataFrame):
+        assert_geodataframe_equal(tested_feature, expected_feature)
```

Comment:
in the snippet of code that you removed, you can see how we can compare `GeoDataFrames` so that equivalent definitions of CRS are ok. That would probably be better (IO sometimes changes the CRS def to an equivalent one)r.